h5netcdf/core.py

```diff
@@ -125,28 +132,61 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class lazy_objects(Mapping):
+    def __init__(self, object_cls):
+        self._objects = list()
+        self._loaded_objects = dict()
+        self._object_cls = object_cls
+        return
```

Comment:
You don't need all these blank `return` statements. Python does that by default.